### DISCUSSION TITLE
Using Apple Payment

### DISCUSSION THREAD

**Discussion by yarodevuci** [braintree/braintree_ios#1105] [General]
We have read in the documentation that it is recommended to collect billing address information, including the billing postal code, and pass that billing postal code with all Apple Pay transactions as a best practice for discount qualification.

Could someone provide guidance on where this billing zip code data should be passed? Should it be handled on the mobile side within our iOS app, or is it something that needs to be processed on the server side?

We added into paymentRequest
`paymentRequest.requiredBillingContactFields = [.postalAddress]
`

Additionally, if there are any specific fields or parameters that we should use for this purpose or any other best practices related to this matter, we would greatly appreciate your insights and recommendations.
@scannillo

**Comment by scannillo** [comment]
👋 Hello @yarodevuci - thank you for using the Braintree iOS sDK! This is mentioned in our developer docs [here](https://developer.paypal.com/braintree/docs/guides/apple-pay/client-side/ios/v5#create-a-pkpaymentrequest). 

For help troubleshooting your integration, please contact Developer Support ([here](https://developer.paypal.com/braintree/help)). For future bugs or issues you find in the SDK, please open an issue [here](https://github.com/braintree/braintree_ios/issues/new/choose) in our repo.
